packages/client/src/components/UploadForm.tsx

```diff
@@ -1,96 +1,158 @@
+import { RequestStatus } from '@/types';
+import { InfoMessage } from '@/components/InfoMessage';
 
-export const UploadForm = () => {
-	const [status, setStatus] = useState<boolean | undefined>(undefined);
-	const auth = useContext(AuthContext);
-	const token = auth.token;
+const uploadFileAndTranscribe = async (file: File, token: string) => {
+	const blob = new Blob([file as BlobPart]);
 
-	if (!token) {
-		return <p>Cannot upload - missing auth token</p>;
+	const response = await authFetch(`/api/signed-url`, token);
+	if (!response) {
+		console.error('Failed to fetch signed url');
+		return false;
 	}
 
-	const uploadFileAndTranscribe = async (fileInput: HTMLInputElement) => {
-		const files = fileInput.files;
-		if (files == undefined || files.length === 0 || !files[0]) {
-			return false;
-		}
-		const file = files[0];
-		const blob = new Blob([file as BlobPart]);
+	const body = SignedUrlResponseBody.safeParse(await response.json());
+	if (!body.success) {
+		console.error('response from signedUrl endpoint in wrong shape');
+		return false;
+	}
 
-		const response = await authFetch(`/api/signed-url`, token);
-		if (!response) {
-			console.error('Failed to fetch signed url');
-			return false;
-		}
+	const uploadStatus = await uploadToS3(body.data.presignedS3Url, blob);
+	if (!uploadStatus) {
+		console.error('Failed to upload to s3');
+		return false;
+	}
 
-		const body = SignedUrlResponseBody.safeParse(await response.json());
-		if (!body.success) {
-			console.error('response from signedUrl endpoint in wrong shape');
-			return false;
-		}
+	const sendMessageResponse = await authFetch('/api/transcribe-file', token, {
+		method: 'POST',
+		headers: {
+			'Content-Type': 'application/json',
+		},
+		body: JSON.stringify({
+			s3Key: body.data.s3Key,
+			fileName: file.name,
+		}),
+	});
+	const sendMessageSuccess = sendMessageResponse.status === 200;
+	if (!sendMessageSuccess) {
+		console.error('Failed to call transcribe-file');
+		return false;
+	}
+	return true;
+};
 
-		const uploadStatus = await uploadToS3(body.data.presignedS3Url, blob);
-		if (!uploadStatus) {
-			console.error('Failed to upload to s3');
-			return false;
-		}
+export const UploadForm = () => {
+	const [status, setStatus] = useState<RequestStatus>(RequestStatus.Ready);
+	const [errorMessage, setErrorMessage] = useState<string | undefined>();
+	const { token } = useContext(AuthContext);
 
-		const sendMessageResponse = await authFetch('/api/transcribe-file', token, {
-			method: 'POST',
-			headers: {
-				'Content-Type': 'application/json',
-			},
-			body: JSON.stringify({
-				s3Key: body.data.s3Key,
-				fileName: file.name,
-			}),
-		});
-		const sendMessageSuccess = sendMessageResponse.status === 200;
-		if (!sendMessageSuccess) {
-			console.error('Failed to call transcribe-file');
-			return false;
-		}
-		return true;
-	};
+	if (!token) {
+		return (
+			<InfoMessage message={'Login required'} status={RequestStatus.Failed} />
+		);
+	}
```

Comment:
can token ever be undefined here, given we're checking it in `template.tsx`? do we just need to convince typescript of this?

Reply:
At the moment it can be undefined because in the AuthState type in template.tsx it is initialised as undefined. I think it would be good to change that though as otherwise every page we add will need to do this additional check. Something for another bit of work/PR though I think